sound/soc/sof/intel/hda-dai.c

```diff
@@ -309,6 +331,23 @@ static int __maybe_unused hda_dai_trigger(struct snd_pcm_substream *substream, i
 			dev_err(sdev->dev, "%s: failed to clean up link DMA\n", __func__);
 			return ret;
 		}
+
```

Comment:
I wonder @ranj063 @ujfalusi if we should modify the logic in https://github.com/thesofproject/linux/commit/ab5593793e9088abcddce30ba8e376e31b7285fd and delay the hda_link_dma_cleanup() on PCM_TRIGGER_STOP. At this point, FW pipeline is put to PAUSED state so it is still hanging on to the link DMA channel (see annotated log in https://github.com/thesofproject/linux/issues/5198#issuecomment-2388354589 ). 

This now works in the original case (same stream restarted later with a new prepare), but indeed if another stream is allocated on host side (like the bug this PR is address), the link DMA channel is allocated to other purposes and the flow won't work anymore.

So could this be delayed to either close() or new prepare() (whichever comes first). For suspend this needs to be immediate.

Reply:
It was delayed before ab5593793e90 and we needed to move it ahead to fix metallic noise (on restart w/o stop) and the explosion of the delay reporting.

Reply:
right, you mean the hda_link_dma_cleanup() ? I think we have tried that but it was causing other issues.